### ISSUE TITLE
Problem with a symmetric group presentation

### ISSUE THREAD

**Issue by MTWhyte** [libsemigroups/libsemigroups#404]
There is a problem with one of the symmetric group presentations, mentioned in the paper of Guralnick, Kantor, Kassabov and Lubotzky. They mention they formerly used the following:

<img width="480" alt="image" src="https://user-images.githubusercontent.com/42549861/201079239-a82a1087-617e-4847-a7a2-6ede45918691.png">

This was implemented in #382. I have made a mistake in the test for this, I think since the presentation due to Burnside and Miller is mentioned just before this, in the same paper. The upshot is: the test added in #382 doesn't actually test the presentation that it should.

If we adapt the test added in #382 to use the presentation in question, and take the degree to be four, we would get the following:

```
    LIBSEMIGROUPS_TEST_CASE("ToddCoxeter",
                            "117",
                            "SymmetricGroup(7, Guralnick)",
                            "[todd-coxeter][quick]") {
      auto rg = ReportGuard(REPORT);

      size_t n = 4;
      auto   s = SymmetricGroup(n, author::Guralnick);
      auto p = make<Presentation<word_type>>(s);
      p.alphabet(n);
      presentation::replace_word(p, word_type({}), {n - 1});
      presentation::add_identity_rules(p, n - 1);
      p.validate();

      ToddCoxeter tc(twosided);
      tc.set_number_of_generators(n);
      for (size_t i = 0; i < p.rules.size() - 1; i += 2) {
        tc.add_pair(p.rules[i], p.rules[i + 1]);
      }

      REQUIRE(tc.number_of_classes() == 24);
    }
```

However, this fails with:
```
tests/test-todd-coxeter.cpp:2273: FAILED:
  REQUIRE( tc.number_of_classes() == 24 )
with expansion:
  72 == 24
```
The function clearly does not give a presentation of the symmetric group of degree 4. Getting an output of `p.rules` just before creating the ToddCoxeter instance gives:
```
{ { 0, 0 }, { 3 },
{ 1, 1 }, { 3 },
{ 2, 2 }, { 3 },
{ 0, 1, 0, 1, 0, 1 }, { 3 },
{ 0, 2, 0, 2, 0, 2 }, { 3 },
{ 1, 0, 1, 0, 1, 0 }, { 3 },
{ 1, 2, 1, 2, 1, 2 }, { 3 },
{ 2, 0, 2, 0, 2, 0 }, { 3 },
{ 2, 1, 2, 1, 2, 1 }, { 3 },
{ 0, 1, 2, 0, 1, 2, 0, 1, 2, 0, 1, 2 }, { 3 },
{ 0, 2, 1, 0, 2, 1, 0, 2, 1, 0, 2, 1 }, { 3 },
{ 1, 0, 2, 1, 0, 2, 1, 0, 2, 1, 0, 2 }, { 3 },
{ 1, 2, 0, 1, 2, 0, 1, 2, 0, 1, 2, 0 }, { 3 },
{ 2, 0, 1, 2, 0, 1, 2, 0, 1, 2, 0, 1 }, { 3 },
{ 2, 1, 0, 2, 1, 0, 2, 1, 0, 2, 1, 0 }, { 3 },
{ 0, 3 }, { 0 },
{ 3, 0 }, { 0 },
{ 1, 3 }, { 1 },
{ 3, 1 }, { 1 },
{ 2, 3 }, { 2 },
{ 3, 2 }, { 2 },
{ 3, 3 }, { 3 } }
```

The generator `3` represents the identity. There are six distinct ordered pairs `(i, j)` from the set `{1, 2, 3}`, and similarly six ordered triples `(i, j, k)`, all of which are there. I believe that this output is exactly what I expected it to be, which makes me wonder if the problem is perhaps mathematical.

**Comment by james-d-mitchell** [comment]
Thanks @MTWhyte, this isn’t so much a bug in libsemigroups but more like a bug in the paper :)

**Comment by james-d-mitchell** [comment]
Resolved in v2.4.0
